### DISCUSSION TITLE
请问QuoteReply要如何获取原消息的id?

### DISCUSSION THREAD

**Discussion by LainNetWork** [mamoe/mirai#976] [Q&A] [answered]
版本：mirai-core-jvm: 2.3.2
您好，我在mirai 1.x版本时，通过消息Id缓存消息中的图片，然后在之后通过回复带有图片的消息来将图片发送到图片服务器，但升级到2.0之后，消息id变成了ids，QuoteReply也找不到获取源消息id的api了。。不知道是不是我姿势不正确？

**Comment by Him188** [comment, marked as answer]
可以使用 `ids ` 第一个元素来定位消息

**Reply by LainNetWork** [reply]
不好意思哈，过年回家没带电脑回去，没看到消息。
其实我主要的疑问是如何通过QuoteReply，获取到被回复的消息的ids。。。找了一圈还是没有找到方法

**Reply by Karlatemp** [reply]
`QuoteReply.source`

**Reply by LainNetWork** [reply]
> `QuoteReply.source`

。。。。我说之前测试为什么从Source里获得的id和实际的id不一样，原来是我代码写错了。。已经解决了，感谢
